File: kotlin/task3.kt
fun digitSum(n: Int): Int = n.toString().map { it.digitToInt() }.sum()

fun digitProduct(n: Int): Int = n.toString().map { it.digitToInt() }.reduce { acc, d -> acc * d }

fun main() {
    print("Введите последовательность чисел через пробел: ")
    val nums = readLine()!!.split(" ").map { it.toInt() }

    print("Номера, где сумма цифр < произведения: ")
    for ((index, num) in nums.withIndex()) {
        if (digitSum(num) < digitProduct(num)) {
            print("${index + 1} ")
        }
    }
    println()
}

File: js/task3.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function digitSum(n) {
  return n.toString().split('').reduce((acc, d) => acc + parseInt(d), 0);
}

function digitProduct(n) {
  return n.toString().split('').reduce((acc, d) => acc * parseInt(d), 1);
}

rl.question("Введите числа через пробел: ", (input) => {
  const nums = input.trim().split(" ").map(Number);
  let result = [];

  nums.forEach((num, index) => {
    if (digitSum(num) < digitProduct(num)) {
      result.push(index + 1); // позиции с 1
    }
  });

  console.log("Номера, где сумма цифр < произведения:", result.join(" "));
  rl.close();
});
